packages/react-data-grid/src/Draggable.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { Omit } from './common/types';
+
+type DragHandler = (x: number, y: number) => void;
+
+interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onDrag' | 'onDragEnd'> {
+  onDrag: DragHandler;
+  onDragEnd: DragHandler;
+}
+
+export default function Draggable({ onDrag, onDragEnd, ...props }: Props) {
+  function onMouseDown(event: React.MouseEvent) {
+    if (event.button !== 0) {
+      return;
+    }
+
+    function onMouseMove(event: MouseEvent) {
+      onDrag(event.clientX, event.clientY);
+    }
+
+    function onMouseUp(event: MouseEvent) {
+      window.removeEventListener('mousemove', onMouseMove);
+      window.removeEventListener('mouseup', onMouseUp);
+      onDragEnd(event.clientX, event.clientY);
+    }
+
+    window.addEventListener('mousemove', onMouseMove);
+    window.addEventListener('mouseup', onMouseUp);
+    onDrag(event.clientX, event.clientY);
+  }
+
+  function onTouchStart(event: React.TouchEvent) {
+    function getTouch(event: TouchEvent) {
+      return Array.from(event.changedTouches).find(t => t.identifier === identifier);
+    }
+
+    function onTouchMove(event: TouchEvent) {
+      const touch = getTouch(event);
+      if (touch) {
+        onDrag(touch.clientX, touch.clientY);
+      }
+    }
+
+    function onTouchEnd(event: TouchEvent) {
+      const touch = getTouch(event);
+      if (!touch) return;
+      window.removeEventListener('touchmove', onTouchMove);
```

Comment:
No, imagine this situation:
- start touching on this component with finger 1
- start touching elsewhere on the page with finger 2
- stop touching with finger 2
- Since we listen to `touchend` on the window, the previous listener will be called, but we properly check that we get the touch that started the dragging behavior, so we stop here.
- We'll only remove the listeners associated with the originating touch finger.